saplings/circuits/src/components/forms/ProposeCircuitForm.js

```diff
@@ -14,18 +14,285 @@
  * limitations under the License.
  */
 
-import React from 'react';
+import React, { useEffect, useReducer } from 'react';
 import { MultiStepForm, Step } from './MultiStepForm';
+import { useNodeRegistryState } from '../../state/nodeRegistry';
+import { useLocalNodeState } from '../../state/localNode';
+
+import NodeCard from '../NodeCard';
+
+import { Chips } from '../Chips';
+
+import './ProposeCircuitForm.scss';
+
+const filterNodes = (state, input) => {
+  const filteredNodes = state.availableNodes.filter(node => {
+    if (node.identity.toLowerCase().indexOf(input) > -1) {
```

Comment:
input should probably be made lower case here, versus in the dom event handler.  Otherwise, this function has to assume that the input is lower case to function properly.

Reply:
updated